Sources/Prometheus/MetricTypes/Summary.swift

```diff
@@ -160,6 +166,9 @@ public class PromSummary<NumType: DoubleRepresentable, Labels: SummaryLabels>: P
             self.count.inc(1)
             self.sum.inc(value)
             self.values.append(value)
+            if self.values.count > self.capacity {
+                self.values.remove(at: 0)
```

Comment:
This is `O(n)` and will have poor performance. It would be better to replace the array here with a `RingBuffer` or `Deque` etc...

Reply:
Thanks for pointing this out! I thought about it but was not sure how to do it the right way. To add a dependency just for this case? To write an own implementation?

After your comment, I have chosen the first option, but swift-collections package's tools version is 5.3 and SwiftPrometheus' tools version is 5.2, so they are incompatible.
Probably, it's a bad idea to change the tools version of this package because of this case.

So, again, I'm not sure what is the best solution in this case.

Reply:
Best solution is probably to use NIOs `CircularBuffer` type that's an automatically expanding `RingBuffer`, which also has an `O(1)` `popFirst()`.

When using this, we should initialise the `CircularBuffer` with an initial capacity of `capacity + 1` (or `capacity` and rework the adding to pop before adding) to avoid (re-)allocations and copies.

Reply:
Agree that both pulling in a dependency, and/or increasing tools version for this would be undesirable 😄

Reply:
Thanks for the advice! I didn't know about NIOs `CircularBuffer`.